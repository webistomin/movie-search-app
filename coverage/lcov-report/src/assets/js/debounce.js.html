<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/assets/js/debounce.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/assets/js</a> debounce.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export default function debounce(func, delay) {
  let inDebounce;
  return function <span class="fstat-no" title="function not covered" >debounceAnswer(</span>...args) {
    const context = <span class="cstat-no" title="statement not covered" >this;</span>
<span class="cstat-no" title="statement not covered" >    clearTimeout(inDebounce);</span>
<span class="cstat-no" title="statement not covered" >    inDebounce = setTimeout(<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >func.apply(context, args),</span> delay);</span>
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jan 19 2019 11:48:02 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
