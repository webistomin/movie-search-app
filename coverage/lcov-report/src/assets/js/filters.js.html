<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/assets/js/filters.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/assets/js</a> filters.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/7</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">40x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">40x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">23x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Vue from 'vue';
&nbsp;
const priceFilter = Vue.filter('getFormattedPrice', (value) =&gt; {
  switch (true) {
<span class="branch-0 cbranch-no" title="branch not covered" >    case Math.abs(Number(value)) &gt;= 1.0e+9:</span>
<span class="cstat-no" title="statement not covered" >      return `${Math.round(Math.abs(Number(value)) / 1.0e+9)} billion dollars`;</span>
    case Math.abs(Number(value)) &gt;= 1.0e+6:
      return `${Math.round(Math.abs(Number(value)) / 1.0e+6)} million dollars`;
<span class="branch-2 cbranch-no" title="branch not covered" >    case Math.abs(Number(value)) &gt;= 1.0e+3:</span>
<span class="cstat-no" title="statement not covered" >      return `${Math.round(Math.abs(Number(value)) / 1.0e+3)} thousand dollars`;</span>
<span class="branch-3 cbranch-no" title="branch not covered" >    default:</span>
<span class="cstat-no" title="statement not covered" >      return `${Math.abs(Number(value))} dollars`;</span>
  }
});
&nbsp;
const dateFilter = Vue.filter('getFormattedDate', date =&gt; new Date(Date.parse(date)).toDateString());
&nbsp;
export default {
  priceFilter,
  dateFilter,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 23 2019 11:56:51 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
