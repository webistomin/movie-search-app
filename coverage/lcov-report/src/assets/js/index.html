<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/assets/js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/assets/js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.62% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">59.26% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/27</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="debounce.js"><a href="debounce.js.html">debounce.js</a></td>
	<td data-value="80" class="pic high"><div class="chart"><div class="cover-fill" style="width: 80%;"></div><div class="cover-empty" style="width:20%;"></div></div></td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="filters.js"><a href="filters.js.html">filters.js</a></td>
	<td data-value="62.5" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 62%;"></div><div class="cover-empty" style="width:38%;"></div></div></td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="throttle.js"><a href="throttle.js.html">throttle.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="16" class="abs medium">8/16</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="16" class="abs medium">8/16</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 23 2019 14:02:58 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
