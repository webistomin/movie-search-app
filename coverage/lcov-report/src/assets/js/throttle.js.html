<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/assets/js/throttle.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/assets/js</a> throttle.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/16</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export default function throttle(func, ms) {
  let isThrottled = false;
  let savedArgs = false;
  let savedThis = false;
&nbsp;
  function wrapper(...args) {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (isThrottled) {
<span class="cstat-no" title="statement not covered" >      savedArgs = args;</span>
<span class="cstat-no" title="statement not covered" >      savedThis = this;</span>
<span class="cstat-no" title="statement not covered" >      return;</span>
    }
&nbsp;
    func.apply(this, args);
&nbsp;
    isThrottled = true;
&nbsp;
    setTimeout(<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >      isThrottled = false;</span>
<span class="cstat-no" title="statement not covered" >      if (savedArgs) {</span>
<span class="cstat-no" title="statement not covered" >        wrapper.apply(savedThis, savedArgs);</span>
<span class="cstat-no" title="statement not covered" >        savedArgs = false;</span>
<span class="cstat-no" title="statement not covered" >        savedThis = false;</span>
      }
    }, ms);
  }
&nbsp;
  return wrapper;
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 22 2019 13:24:03 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
