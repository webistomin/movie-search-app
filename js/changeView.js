const buttonRow = document.getElementById('rowView');
const buttonGrid = document.getElementById('gridView');
const moviesList = document.querySelector('.movies__list');

buttonRow.addEventListener('click', () => {
  if (!buttonRow.classList.contains('page-header__button-active')) {
    buttonRow.classList.add('page-header__button--active');
    buttonGrid.classList.remove('page-header__button--active');
  }
  moviesList.classList.remove('movies__list--grid');
  moviesList.classList.add('movies__list--row');
});

buttonGrid.addEventListener('click', () => {
  if (!buttonGrid.classList.contains('page-header__button-active')) {
    buttonGrid.classList.add('page-header__button--active');
    buttonRow.classList.remove('page-header__button--active');
  }
  moviesList.classList.add('movies__list--grid');
  moviesList.classList.remove('movies__list--row');
});
