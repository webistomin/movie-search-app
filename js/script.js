const buttonRow = document.getElementById('rowView');
const buttonGrid = document.getElementById('gridView');
const moviesList = document.querySelector('.movies__list');

buttonRow.addEventListener('click', () => {
  if (!buttonRow.classList.contains('page-header__button-active')) {
    buttonRow.classList.add('page-header__button--active');
    buttonGrid.classList.remove('page-header__button--active');
  }
  moviesList.classList.remove('movies__list--grid');
  moviesList.classList.add('movies__list--row');
});

buttonGrid.addEventListener('click', () => {
  if (!buttonGrid.classList.contains('page-header__button-active')) {
    buttonGrid.classList.add('page-header__button--active');
    buttonRow.classList.remove('page-header__button--active');
  }
  moviesList.classList.add('movies__list--grid');
  moviesList.classList.remove('movies__list--row');
});

const toggler = document.querySelector('.page-header__toggler');
const mainNav = document.querySelector('.main-nav');

toggler.addEventListener('click', () => {
  if (mainNav.classList.contains('main-nav--closed')) {
    mainNav.classList.add('main-nav--opened');
    mainNav.classList.remove('main-nav--closed');

    toggler.classList.add('page-header__toggler--opened');
    toggler.classList.remove('page-header__toggler--closed');
  } else {
    mainNav.classList.remove('main-nav--opened');
    mainNav.classList.add('main-nav--closed');

    toggler.classList.remove('page-header__toggler--opened');
    toggler.classList.add('page-header__toggler--closed');
  }
});

const userToggler = document.querySelector('.page-header__button--arrow');
const userNav = document.querySelector('.page-header__list');

userToggler.addEventListener('click', () => {
  if (userNav.classList.contains('page-header__list--closed')) {
    userNav.classList.add('page-header__list--opened');
    userNav.classList.remove('page-header__list--closed');

    userToggler.classList.add('page-header__button--opened');
    userToggler.classList.remove('page-header__button--closed');
  } else {
    userNav.classList.remove('page-header__list--opened');
    userNav.classList.add('page-header__list--closed');

    userToggler.classList.remove('page-header__button--opened');
    userToggler.classList.add('page-header__button--closed');
  }
});
