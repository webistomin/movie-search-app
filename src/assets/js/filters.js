import Vue from 'vue';

const priceFilter = Vue.filter('getFormattedPrice', (value) => {
  switch (true) {
    case Math.abs(Number(value)) >= 1.0e+9:
      return `${Math.round(Math.abs(Number(value)) / 1.0e+9)} billion dollars`;
    case Math.abs(Number(value)) >= 1.0e+6:
      return `${Math.round(Math.abs(Number(value)) / 1.0e+6)} million dollars`;
    case Math.abs(Number(value)) >= 1.0e+3:
      return `${Math.round(Math.abs(Number(value)) / 1.0e+3)} thousand dollars`;
    default:
      return `${Math.abs(Number(value))} dollars`;
  }
});

const dateFilter = Vue.filter('getFormattedDate', date => new Date(Date.parse(date)).toDateString());

export default {
  priceFilter,
  dateFilter,
};
