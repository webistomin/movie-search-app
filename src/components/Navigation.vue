<template>
  <nav class="main-nav main-nav--closed">
    <ul class="main-nav__list">
      <li class="main__nav__item">
        <a href="latest.html" class="main-nav__link">
          <svg class="main-nav__icon" width="17" height="17">
            <use xlink:href="#icon-calendar"></use>
          </svg>
          Now Playing
        </a>
      </li>
      <li class="main__nav__item">
        <a href="latest.html" class="main-nav__link">
          <svg class="main-nav__icon" width="17" height="17">
            <use xlink:href="#icon-popular"></use>
          </svg>
          Popular
        </a>
      </li>
      <li class="main__nav__item">
        <a href="latest.html" class="main-nav__link">
          <svg class="main-nav__icon" width="17" height="17">
            <use xlink:href="#icon-top"></use>
          </svg>
          Top Rated
        </a>
      </li>
      <li class="main__nav__item">
        <a href="latest.html" class="main-nav__link">
          <svg class="main-nav__icon" width="17" height="17">
            <use xlink:href="#icon-trailer"></use>
          </svg>
          Upcoming
        </a>
      </li>
    </ul>
  </nav>
</template>

<script>
  export default {
    name: 'Navigation',
  };
</script>

<style lang="sass">
  .main-nav
    width: 100%
    top: 68px
    left: 0
    position: absolute
    background-color: $color-brown--secondary
    z-index: 10

    &--closed
      transform: scaleY(0)
      transform-origin: top
      transition: transform 0.3s ease-in-out

    &--opened
      transform: scaleY(1)
      transform-origin: top
      transition: transform 0.3s ease-in-out

    &__list
      padding: 5px 0

    &__link
      box-sizing: border-box
      position: relative
      padding: 15px 0
      display: flex
      align-items: center
      justify-content: center
      font-size: 13px
      font-weight: 400
      color: $color-text
      transition: all 0.3s ease-in-out
      outline: none

      &:hover,
      &:focus
        color: $color-text--secondary
        transition: all 0.3s ease-in-out

        & .main-nav__icon
          fill: $color-text--secondary
          transition: all 0.3s ease-in-out

      &:active
        color: $color-action
        transition: all 0.3s ease-in-out

        & .main-nav__icon
          fill: $color-action
          transition: all 0.3s ease-in-out

      &--active
        color: $color-action

        & .main-nav__icon
          fill: $color-action

    &__icon
      fill: $color-text
      position: absolute
      left: calc(50% - 70px)
      transition: all 0.3s ease-in-out

    @media (min-width: $screen-sm)
      &__list
        padding: 0

    @media (min-width: $screen-md)
      position: static
      transform: none
      min-height: auto

      &__list
        padding: 15px 0

      &__link
        justify-content: flex-start
        padding-left: 67px

      &__icon
        left: 38px
</style>
