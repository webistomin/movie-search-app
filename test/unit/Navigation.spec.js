import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Navigation from '../../src/components/Navigation';
import routes from '../../src/router/';
import { priceFilter, dateFilter } from '../../src/assets/js/filters';

describe('Header.vue', () => {
  let actions;
  let store;
  let state;
  let mutations;
  let getters;
  let router;
  let vm;
  let localVue;

  beforeEach(() => {
    localStorage.clear();
    jest.clearAllMocks();
    jest.resetModules();

    actions = {
      fetchMovieDetails: jest.fn(),
    };

    mutations = {
      setLoadingState: (state, payload) => {
        state.loadingState = payload;
      },
    };

    state = {
      isNavigationOpened: false,
    };

    getters = {
      getNavigationState: state => state.isNavigationOpened,
    };

    store = new Vuex.Store({
      state,
      mutations,
      actions,
      getters,
    });

    localVue = createLocalVue();
    localVue.filter('getFormattedPrice', priceFilter);
    localVue.filter('getFormattedDate', dateFilter);
    localVue.use(Vuex);
    router = routes;
  });

  it('Matches snapshot', () => {
    vm = shallowMount(Navigation, {
      store,
      localVue,
      router,
    }).vm;
    expect(vm.$el).toMatchSnapshot();
  });

  describe('test "getNavigationClass" computed', () => {
    it('return "main-nav--opened" if nav opened', () => {
      vm = shallowMount(Navigation, {
        store,
        localVue,
        router,
      }).vm;
      state.isNavigationOpened = true;
      expect(vm.getNavigationClass).toEqual('main-nav--opened');
    });

    it('return "main-nav--closed" if nav closed', () => {
      vm = shallowMount(Navigation, {
        store,
        localVue,
        router,
      }).vm;
      state.isNavigationOpened = false;
      expect(vm.getNavigationClass).toEqual('main-nav--closed');
    });
  });
});
