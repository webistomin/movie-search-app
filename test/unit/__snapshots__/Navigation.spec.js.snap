// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Matches snapshot 1`] = `
<nav
  class="main-nav main-nav--closed"
  role="navigation"
>
  <navigationlist-stub />
</nav>
`;
